### ISSUE TITLE
Freeing virtual memory with munmap can't be length 0

### ISSUE THREAD

**Issue by boberfly** [septag/sx#2]
Hey @septag 

This is an awesome library I am learning a lot from what you've written here, thank you. I've been bringing this engine to Linux when I get the chance and I was studying on what you've been doing https://github.com/neilogd/Engine (my fork https://github.com/boberfly/Engine/tree/linux ) as the core lib has similar goals, albeit it is using C++14/namespaces but it is fairly minimal orthodox C++ style for the most part.

One thing I was trying to get around with is the freeing of virtual memory without passing down length, and saw that you might not be aware that Linux can't have 0 passed here:

https://github.com/septag/sx/blob/master/src/virtual-alloc.c#L35
https://linux.die.net/man/3/munmap

I think it'll need some book-keeping of the length unfortunately.

Anyways thought you might want to know, unless Linux can now do what Windows does and frees the full length. Cheers!

**Comment by septag** [comment]
Yes. unfortunately, windows and posix APIs does not map very well here.
I don't want to keep the size as a header in allocated memory because it's rather on a lower-level and caller may expect `page_size` alignment of the pointer from the function.
So I may have to provide a size argument to the API and let the caller keep the size var
I post the fix in the next commit with the bunch of other stuff

thanks
